packages/backend/src/models/GroupsModel.ts

```diff
@@ -270,31 +273,81 @@ export class GroupsModel {
         await this.database('groups').where('group_uuid', groupUuid).del();
     }
 
-    async addGroupMember(
+    async maybeAddGroupMember(
         member: GroupMembership,
     ): Promise<GroupMembership | undefined> {
         // This will return undefined if
         // - the group uuid doesn't exist
         // - the user uuid doesn't exist
         // - the user is already a member of the group
         // - the user is not a member of the group's parent organization
-        const {
-            rows: [insertedMembership],
-        } = await this.database.raw(
-            `
-            INSERT INTO group_memberships (group_uuid, user_id, organization_id)
-            SELECT groups.group_uuid, users.user_id, organization_memberships.organization_id
-            FROM groups
-            INNER JOIN organization_memberships ON groups.organization_id = organization_memberships.organization_id
-            INNER JOIN users ON organization_memberships.user_id = users.user_id
-                AND users.user_uuid = :userUuid
-            WHERE groups.group_uuid = :groupUuid
-            ON CONFLICT DO NOTHING
-            RETURNING *
-        `,
-            member,
-        );
-        return insertedMembership === undefined ? undefined : member;
+        try {
+            const members = await this.addGroupMembers(member.groupUuid, [
+                member.userUuid,
+            ]);
+            return members[0];
+        } catch (e) {
+            if (e instanceof NotFoundError) {
+                return undefined;
+            }
+            throw e;
+        }
+    }
+
+    async addGroupMembers(
+        groupUuid: string,
+        memberUuids: string[],
+    ): Promise<GroupMembership[]> {
+        // Check if the group exists
+        const group = await this.database('groups')
```

Comment:
we could have all these queries inside a single transaction, it will not have much impact on this particular query (apart from race conditions when updating groups/users at the same time), but overall I think is a nice thing to have. 

Not a blocker.